common/arg.cpp

```diff
@@ -697,6 +697,13 @@ gpt_params_context gpt_params_parser_init(gpt_params & params, llama_example ex,
             params.n_keep = value;
         }
     ));
+    add_opt(llama_arg(
+        {"--no-context-shift"},
+        format("disables context shift on inifinite text generation (default: %s)", params.ctx_shift ? "disabled" : "enabled"),
+        [](gpt_params & params) {
+            params.ctx_shift = false;
+        }
+    ));
```

Comment:
For now, this works only with the main example. Later when we extend support to other examples, we can list them here:

```suggestion
    ).set_examples({LLAMA_EXAMPLE_MAIN}));
```